hmcts/pre-api
# Fix: Start Live Event Endpoint does not wait for live event to start

### Change description
- Calling `PUT /media-service/live-event/start/{captureSessionId}` does not wait for live event to start (for both AMS and MK implementations). 
- Capture session is now updated to INITIALISING status instead of STANDBY
- Capture session is now not updated with ingest address (since it is not ready at the time of response). 
- Calling `GET /media-service/live-event/{name}` will automatically update the capture session status and ingest url if the capture session has status INITIALISING and the live event has status of 'Running'. The capture session status will be set to STANDBY and the ingestAddress will be updated based on the live event response. 





> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Run tests for both AMS and MK implementations (currently using AMS):
  - Start live event via endpoint. See returned capture session status has status INITIALISING, started_at and started_by_user_id are not null. See ingest_address is null. 
  - On the media service (AMS/MK) see the live event is starting (might have status `Running` if not quick). 
- Call `GET /media-service/live-event/{name}` every few seconds after starting a live event and wait for status to be 'Running'. When you call `GET /capture-session/{id}` you will see the updated status and ingestAddress values. 





> [!IMPORTANT]
> This PR introduces a breaking change. Please notify the Power Platform team @hmcts/pre-rec-evidence-power-platform.

**Breaking change description**
- Not a breaking change but change from how it currently works- as discussed with power platform devs
  - Now you must poll the `GET /media-service/live-event/{name}` endpoint in order to check the live event status once resourceStatus is `Running` the value of `ingestAddress` field should also be filled. Once the value changes to running, the API will automatically update the capture session to have status STANDBY and set the ingestAddress. 
- Flow should do something like this: 
  - Create capture session
  - Call start live event endpoint (handle errors)
  - Poll (every few seconds) get live event by name endpoint with name = capture session id without hyphens. 
  - Once live event resourceState is `Running` the value of `input_rtmp` should be not null
  - ...
